### PR TITLE
replace deprecated cuda calls in libPMacc and PIConGPU

### PR DESCRIPTION
**Pull request by psychocoderHPC**
replace `cudaThreadSynchronize()` (deprecated since cuda4.0) with `cudaDeviceSynchronize()`

### CODE REVIEW

**Comment by PrometheusPi** [comment]
@psychocoderHPC great job!

Could you fix  this deprecated function call also in `thirdParty/cuda_memtest/cuda_memtest.cu` line 146 and `thirdParty/cuda_memtest/cuda_memtest.h` lines 116 and 135  **or** do you want to open a new pull request for that?

**Comment by psychocoderHPC** [comment]
I opened a extra pull request for `cuda_memtest` #759

File: src/libPMacc/include/eventSystem/events/kernelEvents.hpp [hunk 1]
```diff
@@ -43,14 +43,14 @@ namespace PMacc
 #endif
 
 /** Call activate kernel from taskKernel.
- *  If PMACC_SYNC_KERNEL is 1 cudaThreadSynchronize() is called before
+ *  If PMACC_SYNC_KERNEL is 1 cudaDeviceSynchronize() is called before
```
Comment (psychocoderHPC) [inline src/libPMacc/include/eventSystem/events/kernelEvents.hpp]: `PMACC_SYNC_KERNEL` is now `PMACC_BLOCKING_KERNEL`

thx @PrometheusPi

**Comment by PrometheusPi** [comment]
Looks good to me.

**Pull request merged**